### DISCUSSION TITLE
React JS doesn't support Enable3d

### DISCUSSION THREAD

**Discussion by Hammad-hab** [enable3d/enable3d#297] [Help / Questions]
I use THREEjs and I needed a physics engine for it, enable3d seemed like an option but react JS doesn't support it for some reason because whenever I try to run my app it gives the following error in the console 
`physicsLoader.ts:7 Uncaught ReferenceError: Ammo is not defined`

**Comment by yandeu** [comment]
The path of the physicsLoader has to point to the directory where your ammo.js files are.

You can choose between 2 builds. Kripken and MozillaReality in https://github.com/enable3d/enable3d-website/tree/master/src/lib/ammo

**Reply by Hammad-hab** [reply]
```
import React from 'react';
import './App.css';
import { Project, Scene3D, PhysicsLoader } from 'enable3d'

class MainScene extends Scene3D {


  async init() {
    this.renderer.setPixelRatio(1)
    this.renderer.setSize(window.innerWidth, window.innerHeight)
  }

  async preload() {
    // preload your assets here
  }

  async create() {
    // set up scene (light, ground, grid, sky, orbitControls)
    this.warpSpeed()
  }
}

// set your project configs
const config = { scenes: [MainScene] }

// load the ammo.js file from the /lib folder and start the project
PhysicsLoader("/moz/", () => new Project(config))

function App() {

  return <p>Hello</p>
}

export default App;
```


this is my App.tsx code, `/moz/` is a folder in my src which contains the MozillaReality file. The console gives the following error:
`Uncaught SyntaxError: Unexpected token '<' (at ammo.wasm.js:1:1)
physicsLoader.ts:7 Uncaught ReferenceError: Ammo is not defined`

**Reply by yandeu** [reply]
Have you included all 3 files?
Please check your console to see if it loads from the right path.

If you use react, you probably have a public directory? So maybe try something like `/public/moz`?

**Reply by Hammad-hab** [reply]
Yes I have tried including all three files and I have also tried using both webpack and parcel, is there some dependency I am missing?

**Reply by Hammad-hab** [reply]
So  is this an organic problem with no solution?

**Reply by MayorEmpror** [reply]
Every problem has a solution. Have you tried using 'require' instead of 'import'. It worked for me.

**Reply by Hammad-hab** [reply]
I have tried using both require and import, and it seems that it doesn't fix the error

**Reply by MayorEmpror** [reply]
try downloading the `Three.js + 3D Physics` template, https://github.com/enable3d/threejs-ammojs-webpack-example. After that create a React App and download TypeScript via `npm`. Then try to adjust the `Three.js + 3D Physics template` into your react app, hope it helps

**Reply by Hammad-hab** [reply]
It still gives the error mentioned above

**Comment by scenaristeur** [comment]
I had the same problem with Vuejs but i got it working , perharps could you be inspired : 
The loader is at the end of MainContent.vue
[        PhysicsLoader('lib/ammo', () => new Project(](https://github.com/scenaristeur/noos/blob/dbd46ff90b02db7cc61537cebfdba130c33e8159/src/views/MainContent.vue#L54)

and the /lib/ammo in my public folder 

https://github.com/scenaristeur/noos/tree/main/public

when I run build, they are imported in the dist folder https://github.com/scenaristeur/noos/tree/main/dist


hope this could help

working example on gh-pages 
https://scenaristeur.github.io/flipball/
